File: backend/app/api/health.py
from fastapi import APIRouter
from sqlalchemy import Engine

from app.db.session import get_engine
from app.schemas.health import HealthResponse
from app.services.health import check_health

router = APIRouter(tags=["health"])


def _engine() -> Engine:
    return get_engine()


@router.get("/health", response_model=HealthResponse)
def health() -> HealthResponse:
    report = check_health(_engine())
    return HealthResponse(
        status=report.status,
        database=report.database,
        postgis=report.postgis,
        migration=report.migration,
        detail=report.detail,
    )

File: frontend/src/api/health.ts
import { apiFetch } from './client';

export interface HealthResponse {
  status: 'ok' | 'degraded';
  database: 'ok' | 'error';
  postgis: string | null;
  migration: string | null;
  detail: string | null;
}

export function getHealth(): Promise<HealthResponse> {
  return apiFetch<HealthResponse>('/api/v1/health');
}
